packages/console/src/cloud/hooks/use-cloud-api.ts

```diff
@@ -57,3 +60,34 @@ export const useCloudApi = ({ hideErrorToast = false }: UseCloudApiProps = {}):
 
   return api;
 };
+
+/**
+ * This hook is used to request the cloud `tenantAuthRouter` endpoints, with an organization token.
+ */
+export const useAuthedCloudApi = ({ hideErrorToast = false }: UseCloudApiProps = {}): Client<
+  typeof tenantAuthRouter
+> => {
+  const { currentTenantId } = useContext(TenantsContext);
+  const { isAuthenticated, getOrganizationToken } = useLogto();
+  const api = useMemo(
+    () =>
+      new Client<typeof tenantAuthRouter>({
+        baseUrl: window.location.origin,
+        headers: async () => {
+          if (isAuthenticated) {
+            return {
+              Authorization: `Bearer ${
+                (await getOrganizationToken(getTenantOrganizationId(currentTenantId))) ?? ''
+              }`,
+            };
+          }
+        },
+        before: {
+          ...conditional(!hideErrorToast && { error: toastResponseError }),
+        },
+      }),
+    [currentTenantId, getOrganizationToken, hideErrorToast, isAuthenticated]
+  );
+
+  return api;
+};
```

Comment:
Looks like the token is the only difference. Can we pass it in as a parameter instead of re-write a hook method?

Reply:
The router is also different. The new hook returns `tenantAuthRouter`

Reply:
move foward